lab_2/start.py

```diff
@@ -38,6 +43,31 @@
         UNKNOWN_SAMPLES = file_to_read.read().split('[TEXT]')[1:]
 
     EXPECTED = ['de', 'eng', 'lat']
-    RESULT = ''
+    RESULT = []
+
+    text_corpus = []
+    lang_labels = []
+    STOP_WORDS = []
+    KNN = 3
+
+    for text in DE_SAMPLES:
+        text_corpus.append(remove_stop_words(tokenize(text), STOP_WORDS))
+        lang_labels.append('de')
+    for text in EN_SAMPLES:
+        text_corpus.append(remove_stop_words(tokenize(text), STOP_WORDS))
+        lang_labels.append('eng')
+    for text in LAT_SAMPLES:
+        text_corpus.append(remove_stop_words(tokenize(text), STOP_WORDS))
+        lang_labels.append('lat')
+
+    lang_profiles = get_language_profiles(text_corpus, lang_labels)
+    known_vectors = [get_sparse_vector(text, lang_profiles) for text in text_corpus]
+
+    for text in UNKNOWN_SAMPLES:
+        unknown_text = remove_stop_words(tokenize(text), STOP_WORDS)
+        unknown_vector = get_sparse_vector(unknown_text, lang_profiles)
+        prediction = predict_language_knn_sparse(unknown_vector, known_vectors, lang_labels, KNN)
+        RESULT.append(prediction[0])
```

Comment:
can you describe in writing what happens in` start.py`?

Reply:
imports module os and functions from main.py
creates paths
opens and reads different languages text and samples files
creates empty lists of texts (texts corpus), language labels, and stop_words, and also assigns a value to KNN
then tokenizes all of the texts, removes stop words from them and adds them to the texts corpus as well as adds their language labels to the specific list
creates language profiles of these languages and then gets a sparse vector for each of the texts in the texts corpus
and after that it tokenizes the unknown text, removes stop words from it and also gets its sparse vector
and then predicts what language is the unknown text in by comparing all the sparse vectors